### PR TITLE
channel teaser video implemented

### PR DESCRIPTION
**Pull request by SanJSp**
This is a first solution for the Channel Teaser Video Issue #190 

The collabsing toolbar is only shown, if there is no Channel Teaser Video. If there is one, the Video will be embedded as meta object in the Meta Section List and the collapsing toolbar is locked. 

~Currently there ist still one major bug:~

~When the ChannelDetailsFragment is opened and one clicks on the channel, the Video is not shown. It is as if there was no test for the stageStream, so the collapsing toolbar is shown. When you reopen the channel the video is shown. Unfortunately the same problem appears when you dont have a teaser video, so a Channel without a stageStream would still lock the toolbar and show the Video Layout. I (and Ben) have not been able to find a solution for that.~

Furthermore I'm looking forward to UI feedback. Here is what it looks like:

You see **two play buttons**, because openSAP embedded a play button on the thumbnail.

### CODE REVIEW

**Comment by bengelhaupt** [comment]
I was able to fix the issue. The problem was that in the `ChannelDetailsActivity`, no network request was made, so the unmanaged database entry had been used which might not yet have a `stageStream` set.
For this, I had to create the `CollapsingToolbarViewModelActivity` which is the `CollapsingToolbarActivity` inheriting from `ViewModelActivity` instead of `BaseActivity`.
There was also a bit of white space around the video, caused by the RecyclerView padding. I introduced a somewhat messy but short workaround to fix this by applying negative margin on the video.

File: build.gradle [hunk 1]
```diff
@@ -13,15 +13,15 @@ buildscript {
...
+        realmSchemaVersion  = 9
     }
     repositories {
         jcenter()
         google()
         maven { url "https://maven.fabric.io/public" }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.2'
+        classpath 'com.android.tools.build:gradle:3.5.3'
```
Comment (tobiasrohloff) [inline build.gradle]: No dependency updates in feature branches pls.

File: app/src/main/java/de/xikolo/models/migrate/RealmSchemaMigration.java [hunk 2]
```diff
@@ -121,6 +121,12 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
 
             oldVersion++;
         }
+
+        if (oldVersion == 8) {
+            schema.get("Channel").addRealmObjectField("stageStream", schema.get("VideoStream"));
```
Comment (tobiasrohloff) [inline app/src/main/java/de/xikolo/models/migrate/RealmSchemaMigration.java]: Please merge with changes on master.

Comment (SanJSp) [reply app/src/main/java/de/xikolo/models/migrate/RealmSchemaMigration.java]: done

File: app/src/main/java/de/xikolo/controllers/base/CollapsingToolbarViewModelActivity.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        val TAG: String = CollapsingToolbarViewModelActivity::class.java.simpleName
+    }
+
+    @BindView(R.id.toolbar_image)
+    lateinit var imageView: ImageView
+
+    @BindView(appbar)
+    lateinit var appBarLayout: AppBarLayout
+
+    @BindView(collapsing_toolbar)
```
Comment (tobiasrohloff) [inline app/src/main/java/de/xikolo/controllers/base/CollapsingToolbarViewModelActivity.kt]: `appbar` and `collapsing_toolbar` don't start with `R.id`, so where do they come from?

File: app/src/main/java/de/xikolo/controllers/base/CollapsingToolbarViewModelActivity.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    protected fun lockCollapsingToolbar(title: String) {
+        appBarLayout.setExpanded(false, false)
+        val lp = appBarLayout.layoutParams as CoordinatorLayout.LayoutParams
+        lp.height = actionBarHeight + statusBarHeight
+
+        collapsingToolbar.isTitleEnabled = false
+
+        if (toolbar != null) {
+            toolbar!!.title = title
```
Comment (tobiasrohloff) [inline app/src/main/java/de/xikolo/controllers/base/CollapsingToolbarViewModelActivity.kt]: `toolbar?.title = title`

File: app/src/main/java/de/xikolo/controllers/base/CollapsingToolbarViewModelActivity.kt [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            if (resourceId > 0) {
+                result = application.resources.getDimensionPixelSize(resourceId)
+            }
+            return result
+        }
+
+    private val actionBarHeight: Int
+        get() {
+            val styledAttributes = theme.obtainStyledAttributes(
+                intArrayOf(android.R.attr.actionBarSize))
```
Comment (tobiasrohloff) [inline app/src/main/java/de/xikolo/controllers/base/CollapsingToolbarViewModelActivity.kt]: The closing brackets from `obtainStyledAttributes` should be on a new line.

File: app/src/main/java/de/xikolo/models/migrate/RealmSchemaMigration.java [hunk 6]
```diff
@@ -121,7 +121,7 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
 
             oldVersion++;
         }
-
+
```
Comment (tobiasrohloff) [inline app/src/main/java/de/xikolo/models/migrate/RealmSchemaMigration.java]: We don't need these whitespaces in the diff 😅

File: app/src/main/java/de/xikolo/controllers/base/CollapsingToolbarViewModelActivity.kt (hunk 3 above)
Comment (SanJSp) [reply app/src/main/java/de/xikolo/controllers/base/CollapsingToolbarViewModelActivity.kt]: somehow these landed in the imports. Fixed that.

File: app/src/main/java/de/xikolo/models/migrate/RealmSchemaMigration.java (hunk 6 above)
Comment (SanJSp) [reply app/src/main/java/de/xikolo/models/migrate/RealmSchemaMigration.java]: I have no idea why they're in here. In an ancient commit, I accidentally deleted the line, then I fixed it in another commit. As I understand it Github shows that a line has been taken away, but another has been added. Is this my fault or Githubs?

Comment (tobiasrohloff) [reply app/src/main/java/de/xikolo/models/migrate/RealmSchemaMigration.java]: Probably the IDE added the whitespaces, pls remove them and keep the empty line.

Comment (SanJSp) [reply app/src/main/java/de/xikolo/models/migrate/RealmSchemaMigration.java]: Solved

**Pull request merged**